src/main/host/descriptor/mod.rs

```diff
@@ -0,0 +1,417 @@
+use atomic_refcell::AtomicRefCell;
+use std::sync::Arc;
+
+use crate::cshadow as c;
+use crate::utility::event_queue::{EventQueue, EventSource, Handle};
+
+pub mod pipe;
+
+/// An invocation will fail to compile if the provided type is not Send.
+#[cfg(test)]
+fn verify_send<T: Send>() {}
```

Comment:
Nice idea. I experimented with this a little and found a way to do it with an extra trait definition instead. I slightly prefer this way since it's purely static, and makes it easier to put it near the type definition in an idiomatic way. I don't feel *super* strongly about it vs leaving it as-is though.

Example: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=e3a40bbd6bc607b67dbacb9f1c687e3b
```
trait IsSend : Send {}
trait IsSync : Sync {}

struct Foo {}

// Compiles
impl IsSend for Foo {}
impl IsSync for Foo {}

struct Bar {x: *mut i32}
// Doesn't compile
impl IsSend for Bar {}
impl IsSync for Bar {}
```

if desired we could `#[cfg(test)]`-guard the `impl`s as well, though I *think* no code will end up in the final binary anyway. e.g. https://godbolt.org/z/z5n9rr

Reply:
Cool! I definitely like this way better.